src/zarr/store/memory.py

```diff
@@ -28,12 +29,15 @@ def __init__(
         self,
         store_dict: MutableMapping[str, Buffer] | None = None,
         *,
+        path: str = "",
```

Comment:
this store highlights the fact that not all stores have a meaningful `path`.

Reply:
I'm not sure I agree -- if `store_dict` really is an arbitrary mutable mapping, there's no guarantee that we want all IO operations to be scoped to the top-level key space, and that's exactly what the `path` attribute is for.